extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxCurrentContextFactory.java

```diff
@@ -14,23 +18,34 @@
 import io.smallrye.common.vertx.VertxContext;
 import io.vertx.core.Context;
 import io.vertx.core.Vertx;
+import io.vertx.core.impl.ContextInternal;
+import io.vertx.core.impl.ContextLocalImpl;
 
 public class VertxCurrentContextFactory implements CurrentContextFactory {
 
     private static final String LOCAL_KEY_PREFIX = "io.quarkus.vertx.cdi-current-context";
+    // TODO VertxImpl should be allocated AFTER getting here, to make it work!
+    private static final ContextLocalImpl<? extends ContextState> REQUEST_SCOPED_LOCAL_KEY = new ContextLocalImpl<>();
```

Comment:
So the `ContextLocalImpl` can be used as a key for local data? Shouldn't we use `io.vertx.core.spi.context.storage.ContextLocal.get(Context)` instead? I have to admin that I find the API quite confusing...

Reply:
Depends if you wanna have optimized access or not, I suppose. And yes, indeed using this API within quarkus hopefully improve both (that's why I pinged @vietj here)

Reply:
@vietj wdyt? suggestions?

Reply:
I do not recommend using ContextLocalImpl directly